tools/image_processing/yolo/yolo_training.xml

```diff
@@ -0,0 +1,148 @@
+<tool id="yolo_training" name="Perform YOLO training" version="@VERSION@+galaxy0" profile="24.2">
+    <description>with ultralytics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator" />
+    <requirements>
+	<container type="docker" >quay.io/sunyi000/ultralytics-galaxy:latest</container>
+    </requirements>
+    <command detect_errors="aggressive">
+    <![CDATA[
+      export YOLO_CONFIG_DIR='\$HOME/.config/ultralytics' &&
+
+      mkdir -p ./input_images ./input_yolo ./training ./runs ./models output_files && 
+
+      #for $i, $filename in enumerate($input_images):
+          ln -s '$filename' './input_images/${filename.element_identifier}' &&
+      #end for
+	
+      #for $s, $yolo in enumerate($input_yolo):
+          ln -s '$yolo' './input_yolo/${yolo.element_identifier}' &&
```

Comment:
$s not used?

Reply:
Do the names need to match between input_yolos and input_images?

Reply:
yes the names must match.

If input_images and input_yolo are from anylabeling, they definitely match, anylabeling ensures this.

anylabeling outputs json files which have the same file name as the images.

then the json2yolosegment.xml also ensures the output txt, the input_yolo has the same file name as the input json.

Outside a workflow, if users want to use yolo_train alone, then it's users responsibility to make sure these names match.

Reply:
I think that is fine. Maybe mention that in the tool help?